arches/app/src/components/ControlledListManager/TreeRow.vue

```diff
@@ -0,0 +1,286 @@
+<script setup lang="ts">
+import arches from "arches";
+import Cookies from "js-cookie";
+import { computed, inject, ref, watch } from "vue";
+import { useGettext } from "vue3-gettext";
+
+import Button from "primevue/button";
+import InputText from "primevue/inputtext";
+import { useToast } from "primevue/usetoast";
+
+import MoveRow from "@/components/ControlledListManager/MoveRow.vue";
+
+import { createItem, createList, upsertValue } from "@/components/ControlledListManager/api.ts";
+import { ERROR, displayedRowKey, selectedLanguageKey } from "@/components/ControlledListManager/constants.ts";
+import {
+    bestLabel,
+    findNodeInTree,
+    itemAsNode,
+    listAsNode,
+    nodeIsList,
+} from "@/components/ControlledListManager/utils.ts";
+
+import type { Ref } from "vue";
+import type {
+    TreeExpandedKeys,
+    TreeSelectionKeys,
+} from "primevue/tree/Tree";
+import type { TreeNode } from "primevue/treenode";
+import type {
+    ControlledListItem,
+    DisplayedListItemRefAndSetter,
+    MoveLabels,
+    NewControlledListItem,
+} from "@/types/ControlledListManager";
+import type { Language } from "@/types/arches";
+
+const toast = useToast();
+const { $gettext } = useGettext();
+
+const selectedLanguage = inject(selectedLanguageKey) as Ref<Language>;
+
+const tree = defineModel<TreeNode[]>("tree", { required: true });
+const expandedKeys = defineModel<TreeExpandedKeys>("expandedKeys", { required: true });
+const selectedKeys = defineModel<TreeSelectionKeys>("selectedKeys", { required: true });
+const movingItem = defineModel<TreeNode>("movingItem", { required: true });
+const refetcher = defineModel<number>("refetcher", { required: true });
+const nextNewItem = defineModel<NewControlledListItem>("nextNewItem");
+const newLabelFormValue = defineModel<string>("newLabelFormValue", { required: true });
+const newLabelCounter = defineModel<number>("newLabelCounter", { required: true });
+const newListFormValue = defineModel<string>("newListFormValue", { required: true });
+const newListCounter = defineModel<number>("newListCounter", { required: true });
+const filterValue = defineModel<string>("filterValue", { required: true });
+
+const { isMultiSelecting, node, moveLabels } = defineProps<{
+    isMultiSelecting: boolean,
+    moveLabels: MoveLabels,
+    node: TreeNode,
+}>();
+const { setDisplayedRow } = inject(displayedRowKey) as DisplayedListItemRefAndSetter;
+
+// Workaround for autofocusing the new list/label input boxes
+// https://github.com/primefaces/primevue/issues/2397
+const newListInputRef = ref();
+const newLabelInputRef = ref();
+watch(newLabelInputRef, () => {
+    if (newLabelInputRef.value) {
+        newLabelInputRef.value.$el.focus();
+    }
+});
+watch(newListInputRef, () => {
+    if (newListInputRef.value) {
+        newListInputRef.value.$el.focus();
+    }
+});
+
+const rowLabel = computed(() => {
```

Comment:
why return html here?

Reply:
So the search result can be partially bolded using templating. Is there another better way to do this?